membrane_curvature/tests/test_membrane_curvature.py

```diff
@@ -205,3 +206,84 @@ def test_get_z_surface(x_bin, y_bin, x_range, y_range, expected_surface):
                             [0., 200., 150.], [100., 200., 150.], [200., 200., 150.]])
     surface = get_z_surface(dummy_array, x_bin, y_bin, x_range, y_range)
     assert_almost_equal(surface, expected_surface)
+
+
+class TestMembraneCurvature(object):
+    @pytest.fixture()
+    def universe(self):
+        return mda.Universe(GRO_PO4_SMALL, XTC_PO4_SMALL)
+
+    @pytest.fixture()
+    def universe_dummy(self):
+        a = np.array([[0., 0., 150.], [0., 0., 150.], [200., 0., 150.],
+                      [0., 0., 150.], [100., 100., 150.], [200., 100., 150.],
+                      [0., 200., 150.], [100., 200., 150.], [200., 200., 150.]])
+
+        u = mda.Universe(a, n_atoms=9)
+        u.dimensions = [3, 3, 3,  90., 90., 90.]
+
+        return u
+
+    def test_invalid_selection(self, universe):
+        with pytest.raises(ValueError, match=r'Invalid selection'):
+            MembraneCurvature(universe, select='name P')
+
+    def test_grid_bigger_than_simulation_box(self, universe):
+        with pytest.warns(UserWarning):
+            MembraneCurvature(universe, select='name PO4', x_range=(0, 10))
+            MembraneCurvature(universe, select='name PO4', y_range=(0, 10))
```

Comment:
You need to test these two lines separately, otherwise you can get a passing test even if one of them does not raise the warning.

Reply:
`pytest.warns` can also match part of the warnings string, always a good idea to test that the warning looks like what you expect it.